[FLOC-3972] Ensure icloudapis have unicode lists.

sarum90 opened the pull request:
The EBS driver was returning non-unicode compute IDs for live nodes. This resulted in pyrsistent type errors during raw state discovery and ultimately non-convergence.

Added a functional test so we don't have to rely on acceptance tests to catch this issue in the future.

sarum90 commented:
Note on tests, I started getting `RequestLimitExceeded` Errors, but I did see the acceptance tests go further than they had on previous runs, suggesting this does fix some problem. suspicious about the RequestLimitExceeded errors though, especially this late in the day.

myechuri commented:
Looks great! Please proceed once you have a clean test run. Thanks, @sarum90 !

jml commented on flocker/node/agents/test/test_blockdevice.py:
```diff
@@ -5500,6 +5500,15 @@ def test_current_machine_is_live(self):
+            """
+            live_nodes = self.api.list_live_nodes()
+            for x in live_nodes:
+                self.expectThat(type(x), Is(unicode))
```
:question: No change required, but in case you were curious, testtools also supplies `IsInstance` and `AllMatch` matchers

```
self.assertThat(live_nodes, AllMatch(IsInstance(unicode)))
```

sarum90 replied:
much better. seems I have to merge master in anyway so I'm going to opportunistically update to this as well.

sarum90 commented:
waiting at least for http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/ensure-icloudapis-have-unicode-lists-FLOC-3972/job/run_trial_for_ebs_storage_driver_on_Ubuntu_trusty_flocker.node.agents.functional/4/console before I merge

sarum90 commented:
OK, so just putting this here. I'd _like_ to wait for all of the tests, but I think getting this in is important, so I'm only going to wait for the ones I believe to be relevant.

The only change since the previous commit is in make_icloudapi_test. So long as lint and the ebs functional tests pass I have feel I have "sufficient" confidence to merge this.

sarum90 commented:
EBS functional tests and lint have passed on jenkins. I'll keep a close eye on this and revert it if anything goes sideways, but going to merge. I have high confidence that the rest of the tests will pass.
